### PR TITLE
Refactor `reify ef` to `reifyEff`

### PR DESCRIPTION
**Pull request by jaschdoc**
Related to #2594

### CODE REVIEW

File: main/src/library/DelayMap.flix [hunk 1]
```diff
@@ -170,32 +184,46 @@ namespace DelayMap {
...
         filterWithKey((_, v) -> f(v), m)
 
     ///
     /// Returns a map of all mappings `k -> v` in `m` where `(k, v)` satisfies the predicate `f`.
     ///
     /// The function `f` must be pure.
     ///
-    // TODO: Effect polymorphic split?
-    pub def filterWithKey(f: (k, v) -> Bool, m: DelayMap[k, v]): DelayMap[k, v] with Order[k] =
+    // TODO: Effect polymorphic split? Is that even possible since some nodes have to be removed?
```
Comment (magnus-madsen) [inline main/src/library/DelayMap.flix:225]: No, that is why filter was not in the original list.

**Comment by magnus-madsen** [comment]
LazyList also? :)

**Comment by jaschdoc** [comment]
Fixing DemandList currently. I'll take a look at LazyList after. Do you want it all in this PR?

**Comment by magnus-madsen** [comment]
> 
> 
> Fixing DemandList currently. I'll take a look at LazyList after. Do you want it all in this PR?

No that is fine. We can do it separately. Is this one ready?

NB: DemandList might be broken. Not sure if you can do it there.

**Comment by jaschdoc** [comment]
Yeah this one is ready to go.

**Comment by jaschdoc** [comment]
@magnus-madsen did you accidentally close this without merging?

**Comment by magnus-madsen** [comment]
> 
> 
> @magnus-madsen did you accidentally close this without merging? May have been because of the other PR.

The other PR said closes X which automatically closed this PR.

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**